### ISSUE TITLE
A new awesomeWM release?

### ISSUE THREAD

**Issue by timroes** [awesomeWM/awesome#981]
Wouldn't it be a great time to bundle a new release of awesome WM?

The latest release has issues with Intel Skylake architecture making it nearly impossible to use multi monitor setups with awesome WM. So everyone with an modern Intel cpu is forced to use the master branch (in which the issues are fixed).

Since API changed a lot in the master branch, one would need to change their config a lot when switching from latest release to master. Using the master branch leaves it unsure how much more breaking API changes will come, which makes it hard to invest the time to change the config already.

I would really like to see a new awesome release once the new API is considered stable for some time, so one could use awesome wm again on Intel Skylake architecture and using a stable version from the package manager.

Any thoughts on that?

PS: I know I already asked that in IRC, but I really would like not to go with master, since every update could break my whole config again, due to API changes and also my computers then use different awesome versions depending on when I updated master last. But more than that I think it would be great if there would be a version that people with Skylake processors can install, which would just work, since the way it is, you make it hard for that kind of people to switch to awesome WM, because they might not be in the mood to compile master to try out a new (and yet awesome) WM.

**Comment by psychon** [comment]
Duplicate of #553 (kind of) and #726. Overview of current issues is the 3.6 milestone: https://github.com/awesomeWM/awesome/milestones/v3.6

I guess the biggest "problem" are issues like #976. With this I mean "everything which breaks due to awesome no longer restarting on monitor changes". Do we want to give this some time so that more issues are fixed or do we e.g. add an FAQ entry "this is how you make awesome restart on screen changes again" as a work-around for issues?

> The latest release has issues with Intel Skylake architecture

Huh? Which one? How was it fixed? Can the fix be backported to the 3.5 branch?

**Comment by timroes** [comment]
Okay as long as there is still a fixed amount of issues which should be finished before the next release, I would totally wait for these.

Regarding the Skylake architecture: Skylake currently makes a lot of problems in Linux. The kernel support is yet far from complete. Just the last version fixed the problem, that I can close my laptop again without the Kernel freezing (due to unssuported pstates; workaround was to disable pstates via kernel parameter). If you have virtualization turned on in BIOS and plug in an HDMI cable the xserver sometimes freezes, etc.

In general it seems that also the graphic part has large problems with the new skylake architecture. E.g. if I have compton enabled, one chrome tab with Google Music open will result in 4 cores at 100% load. Killing compton solves that problem.

So much for the general summary. What's the problem of awesomeWM? [...]

I just tried to make you a screenshot of the problem and as it seems it no longer exists due to some update of other packages. The problem was: with two displays with different resolution in cloned mode or just one display, but with a resolution that isn't its optimal resolution, will cause the image to draw weird. The actual picture was shifted by around 100 pixels to the bottom and the above layer was the upper 100 pixels of the screen before one changed the resolution (or monitors). But still the mouse moved over all the screen area and you actualy had to click on the actual positions (not the shifted once, that are drawn).

This was reproducible on my computer and the same behaviour wasn't with the same awesome wm (and OS) on another one withouh Skylake. I updated the Skylake to awesome master and the problem vanished. Switching back to awesome 3.5.9 brang it back. But apparently it was a problem in another library (that the awesome-git package might udpated too) or in the communication between awesome and some other library, that has now updated in Archlinux. At least I am not able to reproduce it anymore on this system since todays update (on Friday I could still reproduce it).

So since this problem is now gone, I also don't have any hurry in updating anymore, since I have a stable working awesome release again :-) Sorry for the quite unnecessary issue and thanks again for your great work!
